bin/rucio

```diff
@@ -1796,6 +1795,91 @@ def list_datasets_rse(args):
     return SUCCESS
 
 
+@exception_handler
+def add_lifetime_exception(args):
+    """
+    %(prog)s add_lifetime_exception [options] <field1=value1 field2=value2 ...>
+
+    Declare a list of bad replicas.
+
+    """
+    client = get_client(args)
+    if not args.reason:
+        logger.error('reason for the extension is mandatory')
+        return FAILURE
+    reason = args.reason
+    if not args.expiration:
+        logger.error('expiration is mandatory')
+        return FAILURE
+    try:
+        expiration = isoparse(args.expiration)
+    except ValueError as err:
+        logger.error(err)
+        return FAILURE
+
+    if not args.inputfile:
+        logger.error('inputfile is mandatory')
+        return FAILURE
+    with open(args.inputfile) as infile:
+        dids = list(filter(None, [line.strip() for line in infile]))
+
+    dids_list = []
+    containers = []
+    datasets = []
+    error_types = ['Total DIDs', 'DID not submitted because it is a file', 'DID not submitted because it is a container', 'DID not submitted because it is not part of the lifetime campaign', 'DID successfully submitted']
+    for did in dids:
+        scope, name = get_scope(did, client)
+        dids_list.append({'scope': scope, 'name': name})
+    summary = {0: len(dids_list), 1: 0, 2: 0, 3: 0, 4: 0}
+    for meta in client.get_metadata_bulk(dids_list):
+        scope, name = meta['scope'], meta['name']
+        dids_list.remove({'scope': scope, 'name': name})
+        if meta['did_type'] == 'FILE':
+            logger.warning('%s:%s is a file. Will be ignored' % (scope, name))
+            summary[1] += 1
+        elif meta['did_type'] == 'CONTAINER':
+            logger.warning('%s:%s is a container. It needs to be resolved' % (scope, name))
+            containers.append({'scope': scope, 'name': name})
+            summary[2] += 1
+        elif not meta['eol_at']:
+            logger.warning('%s:%s is not affected by the lifetime model' % (scope, name))
+            summary[3] += 1
+        else:
+            logger.info('%s:%s will be declared' % (scope, name))
+            datasets.append({'scope': scope, 'name': name})
+            summary[4] += 1
+
+    for did in dids_list:
+        scope = did['scope']
+        name = did['name']
+        logger.warning('%s:%s does not exist' % (scope, name))
```

Comment:
I also just noticed, that this code is never reached.

Reply:
Okay, this comment could be wrong. I'm not sure if `client.get_metadata_bulk` always returns a list with the same length as the input list. If not, it should be safe to ignore.